ipa-core/src/protocol/dp/distributions.rs

```diff
@@ -63,44 +58,92 @@ impl Distribution<f64> for RoundedBoxMuller {
         self.inner.sample(rng).round()
     }
 }
-
 impl From<BoxMuller> for RoundedBoxMuller {
     fn from(value: BoxMuller) -> Self {
         Self { inner: value }
     }
 }
-
+/// Double Geometric Distribution
+///
+/// Generates a sample from a geometric distribution with the given success probability.
+fn generate_geometric<R: Rng + ?Sized>(probability: f64, rng: &mut R) -> isize {
+    // Create a Bernoulli distribution with the specified success probability
+    let bernoulli = Bernoulli::new(probability).expect("Invalid probability");
+    // Generate Bernoulli random numbers until the first success
+    // let mut rng = rand::thread_rng();
+    let mut attempts = 0;
+    // let sample = ud.sample(rng);
+    while !bernoulli.sample(rng) {
+        attempts += 1;
+    }
+    attempts
+}
+/// Generates a sample from a double geometric distribution with the given success probability and shift parameter.
+fn generate_double_geometric<R: Rng + ?Sized>(s: f64, shift: isize, rng: &mut R) -> isize {
+    let success_probability = 1.0 - E.powf(-1.0 / s);
+    let attempts1 = generate_geometric(success_probability, rng);
+    let attempts2 = generate_geometric(success_probability, rng);
+    shift + attempts1 - attempts2
+}
+/// Generates a sample from a double geometric distribution with the given success probability and shift parameter.
```

Comment:
code style suggestion (I don't know why format is not enforcing it) to leave a blank line between declarations - makes it a bit easier to read